frontend/src/store/modules/page_views/actions.js

```diff
@@ -16,4 +40,52 @@ export default {
             context.commit(RESET_BUTTON_FETCHING, data.button);
         }
     },
+    [FETCH_PAGE_DATA]: (context) => {
+        const time = getTimeByPeriod(context.state.selectedPeriod);
```

Comment:
not quite sure that you need to define time for next two actions here, each of this actions can do this on they own